compaction/time_window_compaction_strategy.cc

```diff
@@ -361,7 +361,7 @@ time_window_compaction_strategy::get_sstables_for_compaction(compaction_group_vi
         clogger.debug("[{}] TWCS skipping check for fully expired SSTables", fmt::ptr(this));
     }
 
-    auto compaction_candidates = get_next_non_expired_sstables(table_s, control, std::move(candidates), compaction_time);
+    auto compaction_candidates = get_next_non_expired_sstables(table_s, control, std::move(candidates), compaction_time, state);
```

Comment:
alternatively manager could have fed compaction_strategy_state_ptr into get_sstables_for_compaction() and the strategy is responsible for performing the conversion on a stable ptr. But I think this is good enough for now.